### ISSUE TITLE
Eclipse takes a very long time to initialize (seems to hang on startup) loading feature metadata for Liberty Server(s).

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/open-liberty-tools#493]
With Eclipse + LDT versions:

* Eclipse Platform	4.29.0.v20230903-1000	org.eclipse.platform.feature.group	Eclipse.org
*  WebSphere® Application Server Liberty Tools	2023.0.200.v20231128_1051	com.ibm.websphere.wdt.server.tools.main.feature.group	IBM

In a previous session, I had previously created a couple of Runtime Environments / Servers.    

Later, Eclipse seemed to hang on startup.   This happened with both a Semeru Open J9 and Hotspot-based Java 17 JDK.

I took a java dump (for Semeru) (sorry, IBM-internal link [here](https://gist.github.ibm.com/skurz/437d149b41cfe13c36bb508d66895186)).

It's not actually "hung" or deadlocked in a strict sense at this point but it is in a loop that is going to take a long time to finish...

So we're looping through and doing this for each feature in the runtime... and waiting 6 minutes for each.

```
FeatureListCoreMetadata(Metadata<PayloadType>).getGeneratedInfo(IMetadataGenerator) line: 124	
FeatureListCoreMetadata(AbstractFeatureListMetadata).getFeatureListMaps(IMetadataGenerator) line: 39	
FeatureList.getFeatureMap(WebSphereRuntime) line: 303	
FeatureList.getFeature(String, WebSphereRuntime) line: 327	
FeatureList.getFeatureAPIJars(String, WebSphereRuntime) line: 230	
WebSphereRuntimeClasspathHelper.findFeaturesPerJar() line: 74	
```

It seems like the reason it's not exiting out of the loop is that the [com.ibm.ws.st](http://com.ibm.ws.st/).core.internal.generation.GeneratorJob.run(IProgressMonitor)  gets scheduled through this call hierarchy, but it never actually gets to run

(Here's a Call Hierarchy screenshot).

![image](https://github.com/OpenLiberty/open-liberty-tools/assets/4081634/843deb9d-a967-4fd9-b927-ed22eeb9ece2)

**Comment by pnickoll** [comment]
Think fix #500 should fix this

**Comment by mattcolegate** [comment]
Tested yesterday with latest build and was not able to recreate - I believe this issue has been resolved by the fix mentioned, so I'm closing the issue. Please feel free to re-open if you stll have this issue using latest development build.
